core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -55,23 +56,24 @@ public static FileOperations getInstance() {
    */
 
   public abstract FileSKVIterator openReader(String file, Range range, Set<ByteSequence> columnFamilies, boolean inclusive, FileSystem fs, Configuration conf,
-      AccumuloConfiguration tableConf) throws IOException;
+      RateLimiter readLimiter, AccumuloConfiguration tableConf) throws IOException;
 
   public abstract FileSKVIterator openReader(String file, Range range, Set<ByteSequence> columnFamilies, boolean inclusive, FileSystem fs, Configuration conf,
-      AccumuloConfiguration tableConf, BlockCache dataCache, BlockCache indexCache) throws IOException;
+      RateLimiter readLimiter, AccumuloConfiguration tableConf, BlockCache dataCache, BlockCache indexCache) throws IOException;
 
   /**
    * Open a reader that fully support seeking and also enable any optimizations related to seeking, like bloom filters.
    *
    */
 
-  public abstract FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, AccumuloConfiguration acuconf)
-      throws IOException;
+  public abstract FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, RateLimiter readLimiter,
```

Comment:
> Somehow get the RateLimiterFactory directly from the config? That'd be a stop-gap.

@joshelser Not sure exactly what you mean, but what I meant was along the lines of creating a "FileOptions" (or similar) object to pass around instead of the AccumuloConfiguration object which is narrowly applicable to the context of those utility methods (context: operations on files?), and which is composed of the relevant configs from wherever they are sourced.

@ShawnWalker I would consider it an optional improvement to this pull request. If the code modified here can be clearly demarcated as a distinct context, it may benefit from this kind of improvement as an alternative to adding the extra parameter everywhere. It can also be done at some future point in time, as a more general improvement (and a much bigger one... as it would require identifying multiple distinct contexts instead of possibly doing one at a time).